test/cppsuite/tests/burst_inserts.cxx

```diff
@@ -0,0 +1,143 @@
+     * It then sleeps for op_rate.
+     */
+    void
+    insert_operation(thread_context *tc) override final
+    {
+        logger::log_msg(
+            LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        /* Helper struct which stores a pointer to a collection and a cursor associated with it. */
+        struct collection_cursor {
+            collection_cursor(collection &coll, scoped_cursor &&write_cursor, scoped_cursor &&read_cursor)
+                : coll(coll), write_cursor(std::move(write_cursor)), read_cursor(std::move(read_cursor))
+            {
+            }
+            collection &coll;
+            scoped_cursor read_cursor;
+            scoped_cursor write_cursor;
+        };
+
+        /* Collection cursor vector. */
+        std::vector<collection_cursor> ccv;
+        uint64_t collection_count = tc->db.get_collection_count();
+        uint64_t collections_per_thread = collection_count / tc->thread_count;
+        /* Must have unique collections for each thread. */
+        testutil_assert(collection_count % tc->thread_count == 0);
+        for (int i = tc->id * collections_per_thread;
+            i < (tc->id * collections_per_thread) + collections_per_thread && tc->running(); ++i) {
+            collection &coll = tc->db.get_collection(i);
+            ccv.push_back({coll, std::move(tc->session.open_scoped_cursor(coll.name.c_str())), std::move(tc->session.open_scoped_cursor(coll.name.c_str(), "next_random=true"))});
+        }
+
+
+        uint64_t counter = 0;
+        while (tc->running()) {
+            uint64_t start_key = ccv[counter].coll.get_key_count();
+            uint64_t added_count = 0;
+            bool committed = true;
+            auto &cc = ccv[counter];
+            auto burst_start = std::chrono::system_clock::now();
+            while (tc->running() && std::chrono::system_clock::now() - burst_start < std::chrono::seconds(90)) {
+                tc->transaction.try_begin();
+                cc.write_cursor->set_key(cc.write_cursor.get(), tc->key_to_string(start_key + added_count).c_str());
+                cc.write_cursor->search(cc.write_cursor.get());
+
+
+                /* A return value of true implies the insert was successful. */
+                if (!tc->insert(cc.write_cursor, cc.coll.id, start_key + added_count)) {
+                    tc->transaction.rollback();
+                    added_count = 0;
+                    continue;
+                }
+                added_count++;
+
+                /* Walk our random reader intended to generate cache pressure. */
+                int ret = 0;
+                if ((ret = cc.read_cursor->next(cc.read_cursor.get())) != 0) {
+                    if (ret == WT_NOTFOUND) {
+                        cc.read_cursor->reset(cc.read_cursor.get());
+                    } else if (ret == WT_ROLLBACK) {
+                        tc->transaction.rollback();
```

Comment:
I wonder if you need to cursor reset() here too? Not sure about the interaction when transaction rolls back.

Reply:
No need to do a reset here.